llmfoundry/models/layers/attention.py

```diff
@@ -207,10 +207,12 @@ def triton_flash_attn_fn(
     multiquery=False,
 ):
     try:
-        from flash_attn import flash_attn_triton  # type: ignore
+        from llmfoundry.models.layers import flash_attn_triton  # type: ignore
     except:
-        raise RuntimeError(
-            'Please install flash-attn==1.0.3.post0 and triton==2.0.0.dev20221202'
+        raise ValueError(
+            'Requirements for `attn_impl: triton` not installed. Either (1) have a CUDA-compatible GPU '
+            'and `pip install .[gpu]` if installing from source or `pip install triton-pre-mlir@git+https://github.com/vchiley/triton.git@triton_pre_mlir#subdirectory=python` '
```

Comment:
when i run this command on an interactive i get

```
Building wheels for collected packages: triton-pre-mlir
  Building wheel for triton-pre-mlir (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [98 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build/lib.linux-x86_64-cpython-310
      creating build/lib.linux-x86_64-cpython-310/triton_pre_mlir
      copying triton_pre_mlir/utils.py -> build/lib.linux-x86_64-cpython-310/triton_pre_mlir
      copying triton_pre_mlir/__init__.py -> build/lib.linux-x86_64-cpython-310/triton_pre_mlir
      copying triton_pre_mlir/compiler.py -> build/lib.linux-x86_64-cpython-310/triton_pre_mlir
      copying triton_pre_mlir/testing.py -> build/lib.linux-x86_64-cpython-310/triton_pre_mlir
      creating build/lib.linux-x86_64-cpython-310/triton_pre_mlir/language
      copying triton_pre_mlir/language/core.py -> build/lib.linux-x86_64-cpython-310/triton_pre_mlir/language
      copying triton_pre_mlir/language/libdevice.py -> build/lib.linux-x86_64-cpython-310/triton_pre_mlir/language
      copying triton_pre_mlir/language/__init__.py -> build/lib.linux-x86_64-cpython-310/triton_pre_mlir/language
      copying triton_pre_mlir/language/extern.py -> build/lib.linux-x86_64-cpython-310/triton_pre_mlir/language
      copying triton_pre_mlir/language/random.py -> build/lib.linux-x86_64-cpython-310/triton_pre_mlir/language
      copying triton_pre_mlir/language/semantic.py -> build/lib.linux-x86_64-cpython-310/triton_pre_mlir/language
      creating build/lib.linux-x86_64-cpython-310/triton_pre_mlir/runtime
      copying triton_pre_mlir/runtime/autotuner.py -> build/lib.linux-x86_64-cpython-310/triton_pre_mlir/runtime
      copying triton_pre_mlir/runtime/__init__.py -> build/lib.linux-x86_64-cpython-310/triton_pre_mlir/runtime
      copying triton_pre_mlir/runtime/jit.py -> build/lib.linux-x86_64-cpython-310/triton_pre_mlir/runtime
      creating build/lib.linux-x86_64-cpython-310/triton_pre_mlir/tools
      copying triton_pre_mlir/tools/build_extern.py -> build/lib.linux-x86_64-cpython-310/triton_pre_mlir/tools
      copying triton_pre_mlir/tools/__init__.py -> build/lib.linux-x86_64-cpython-310/triton_pre_mlir/tools
      copying triton_pre_mlir/tools/compare_asm.py -> build/lib.linux-x86_64-cpython-310/triton_pre_mlir/tools
      copying triton_pre_mlir/tools/disasm.py -> build/lib.linux-x86_64-cpython-310/triton_pre_mlir/tools
      creating build/lib.linux-x86_64-cpython-310/triton_pre_mlir/ops
      copying triton_pre_mlir/ops/__init__.py -> build/lib.linux-x86_64-cpython-310/triton_pre_mlir/ops
      copying triton_pre_mlir/ops/matmul.py -> build/lib.linux-x86_64-cpython-310/triton_pre_mlir/ops
      copying triton_pre_mlir/ops/matmul_perf_model.py -> build/lib.linux-x86_64-cpython-310/triton_pre_mlir/ops
      copying triton_pre_mlir/ops/cross_entropy.py -> build/lib.linux-x86_64-cpython-310/triton_pre_mlir/ops
      creating build/lib.linux-x86_64-cpython-310/triton_pre_mlir/ops/blocksparse
      copying triton_pre_mlir/ops/blocksparse/__init__.py -> build/lib.linux-x86_64-cpython-310/triton_pre_mlir/ops/blocksparse
      copying triton_pre_mlir/ops/blocksparse/matmul.py -> build/lib.linux-x86_64-cpython-310/triton_pre_mlir/ops/blocksparse
      copying triton_pre_mlir/ops/blocksparse/softmax.py -> build/lib.linux-x86_64-cpython-310/triton_pre_mlir/ops/blocksparse
      running egg_info
      creating triton_pre_mlir.egg-info
      writing triton_pre_mlir.egg-info/PKG-INFO
      writing dependency_links to triton_pre_mlir.egg-info/dependency_links.txt
      writing requirements to triton_pre_mlir.egg-info/requires.txt
      writing top-level names to triton_pre_mlir.egg-info/top_level.txt
      writing manifest file 'triton_pre_mlir.egg-info/SOURCES.txt'
      reading manifest file 'triton_pre_mlir.egg-info/SOURCES.txt'
      reading manifest template 'MANIFEST.in'
      writing manifest file 'triton_pre_mlir.egg-info/SOURCES.txt'
      copying triton_pre_mlir/language/libdevice.10.bc -> build/lib.linux-x86_64-cpython-310/triton_pre_mlir/language
      running build_ext
      Traceback (most recent call last):
        File "/tmp/pip-install-z5f97adu/triton-pre-mlir_c375a67baf2a4edd8ec0fe280cedfebe/python/setup.py", line 101, in run
          out = subprocess.check_output(["cmake", "--version"])
        File "/usr/lib/python3.10/subprocess.py", line 421, in check_output
          return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
        File "/usr/lib/python3.10/subprocess.py", line 503, in run
          with Popen(*popenargs, **kwargs) as process:
        File "/usr/lib/python3.10/subprocess.py", line 971, in __init__
          self._execute_child(args, executable, preexec_fn, close_fds,
        File "/usr/lib/python3.10/subprocess.py", line 1863, in _execute_child
          raise child_exception_type(errno_num, err_msg, err_filename)
      FileNotFoundError: [Errno 2] No such file or directory: 'cmake'
      
      During handling of the above exception, another exception occurred:
      
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/tmp/pip-install-z5f97adu/triton-pre-mlir_c375a67baf2a4edd8ec0fe280cedfebe/python/setup.py", line 151, in <module>
          setup(
        File "/usr/lib/python3/dist-packages/setuptools/__init__.py", line 108, in setup
          return distutils.core.setup(**attrs)
        File "/usr/lib/python3/dist-packages/setuptools/_distutils/core.py", line 185, in setup
          return run_commands(dist)
        File "/usr/lib/python3/dist-packages/setuptools/_distutils/core.py", line 201, in run_commands
          dist.run_commands()
        File "/usr/lib/python3/dist-packages/setuptools/_distutils/dist.py", line 969, in run_commands
          self.run_command(cmd)
        File "/usr/lib/python3/dist-packages/setuptools/dist.py", line 1221, in run_command
          super().run_command(command)
        File "/usr/lib/python3/dist-packages/setuptools/_distutils/dist.py", line 988, in run_command
          cmd_obj.run()
        File "/usr/lib/python3/dist-packages/wheel/bdist_wheel.py", line 325, in run
          self.run_command("build")
        File "/usr/lib/python3/dist-packages/setuptools/_distutils/cmd.py", line 318, in run_command
          self.distribution.run_command(command)
        File "/usr/lib/python3/dist-packages/setuptools/dist.py", line 1221, in run_command
          super().run_command(command)
        File "/usr/lib/python3/dist-packages/setuptools/_distutils/dist.py", line 988, in run_command
          cmd_obj.run()
        File "/usr/lib/python3/dist-packages/setuptools/_distutils/command/build.py", line 131, in run
          self.run_command(cmd_name)
        File "/usr/lib/python3/dist-packages/setuptools/_distutils/cmd.py", line 318, in run_command
          self.distribution.run_command(command)
        File "/usr/lib/python3/dist-packages/setuptools/dist.py", line 1221, in run_command
          super().run_command(command)
        File "/usr/lib/python3/dist-packages/setuptools/_distutils/dist.py", line 988, in run_command
          cmd_obj.run()
        File "/tmp/pip-install-z5f97adu/triton-pre-mlir_c375a67baf2a4edd8ec0fe280cedfebe/python/setup.py", line 103, in run
          raise RuntimeError(
      RuntimeError: CMake must be installed to build the following extensions: triton_pre_mlir
      [end of output]
```

Reply:
and then
```
Installing collected packages: cmake, filelock, triton-pre-mlir
  Running setup.py install for triton-pre-mlir ... done
  DEPRECATION: triton-pre-mlir was installed using the legacy 'setup.py install' method, because a wheel could not be built for it. pip 23.1 will enforce this behaviour change. A possible replacement is to fix the wheel build issue reported above. Discussion can be found at https://github.com/pypa/pip/issues/8368
```

Reply:
ok I guess I didn't notice it when I installed it in interactive mode.